src/kOS/Suffixed/Timespan.cs

```diff
@@ -128,48 +129,31 @@ public double ToUnixStyleTime()
 
-        public override object TryOperation(string op, object other, bool reverseOrder)
+        public override bool Equals(object obj)
         {
-            other = ConvertToDoubleIfNeeded(other);
-
-            // Order shouldn't matter here
-            if (other is TimeSpan && op == "+") return this + (TimeSpan)other;
-            if (other is double && op == "+") return this + (double)other;
-            if (other is double && op == "*") return this * (double)other; // Order would matter here if this were matrices
-
-            if (!reverseOrder)
+            Type compareType = typeof(TimeSpan);
+            if (compareType.IsInstanceOfType(obj))
             {
-                if (other is TimeSpan && op == "-") return this - (TimeSpan)other;
-                if (other is TimeSpan && op == "/") return this / (TimeSpan)other;
-                if (other is double && op == "-") return this - (double)other;
-                if (other is double && op == "/") return this / (double)other;
-                if (other is TimeSpan && op == ">") return this > (TimeSpan)other;
-                if (other is TimeSpan && op == "<") return this < (TimeSpan)other;
-                if (other is TimeSpan && op == ">=") return this >= (TimeSpan)other;
-                if (other is TimeSpan && op == "<=") return this <= (TimeSpan)other;
-                if (other is double && op == ">") return this > (double)other;
-                if (other is double && op == "<") return this < (double)other;
-                if (other is double && op == ">=") return this >= (double)other;
-                if (other is double && op == "<=") return this <= (double)other;
+                TimeSpan t = obj as TimeSpan;
+                // Check the equality of the span value
+                return span == t.ToUnixStyleTime();
             }
-            else
-            {
-                if (other is TimeSpan && op == "-") return (TimeSpan)other - this;
-                if (other is TimeSpan && op == "/") return (TimeSpan)other / this;
-                if (other is double && op == "-") return (double)other - this;
-               if (other is double && op == "/") return (double)other / this; // Can't imagine why the heck you'd want to do this but here it is
-               if (other is TimeSpan && op == ">") return this < (TimeSpan)other;
-               if (other is TimeSpan && op == "<") return this > (TimeSpan)other;
-               if (other is TimeSpan && op == ">=") return (TimeSpan)other >= this;
-               if (other is TimeSpan && op == "<=") return (TimeSpan)other <= this;
-               if (other is double && op == ">") return (double)other > this;
-               if (other is double && op == "<") return (double)other < this;
-               if (other is double && op == ">=") return (double)other >= this;
-               if (other is double && op == "<=") return (double)other <= this;
+            return false;
```

Comment:
Okay, after some testing, I concluded that it wouldn't work anyway.  The double timestamp is so precise that even if you print the ``:seconds`` suffix of the timestamp, then directly type in exactly what you see on screen as a double (so it should be ==), the comparison claims the timestamp is greater than the double.  So the mere tiny rounding that happens when printing the value as a decimal is enough to make equality fail, so it would need an epsilon in the comparison and that's just getting wonky.  It's better to force people to do their own epsilon logic with greater than and less than.